### ISSUE TITLE
Compile fail on singleton type via Shapeless

### ISSUE THREAD

**Issue by benhutchison** [lihaoyi/Ammonite#356]
Ammonite 0.5.2 / Scala 2.11.7:
```
@ load.ivy("com.chuusai" %% "shapeless" % "2.3.0-RC4")

@ import shapeless._
import shapeless._
@ import syntax.singleton._
import syntax.singleton._
@ "bar".narrow
Compilation Failed
Main.scala:328: exception during macro expansion:
scala.MatchError: String("bar") (of class scala.reflect.internal.Types$UniqueConstantType)
	at ammonite.repl.frontend.TPrintLowPri$.rec0$1(TPrintImpl.scala:144)
	at ammonite.repl.frontend.TPrintLowPri$.typePrintImpl(TPrintImpl.scala:181)

              .print(res4, res4, "res4", scala.None)
                    ^
```

SBT console:
```
scala> import shapeless._
import shapeless._

scala> import syntax.singleton._
import syntax.singleton._

scala> "bar".narrow
res0: String("bar") = bar
```

**Comment by lihaoyi** [comment]
Looks like I fixed it as part of 0.5.5

https://github.com/lihaoyi/Ammonite/commit/74b2e075a2e833c007972b70425de1560157e7c0

I must've forgotten to put it in the changelog

**Comment by lihaoyi** [comment]
```scala
haoyi-mbp:~ haoyi$ ~/amm
Loading...
Welcome to the Ammonite Repl 0.5.5
(Scala 2.11.7 Java 1.8.0_25)
@  load.ivy("com.chuusai" %% "shapeless" % "2.3.0-RC4")
:: loading settings :: url = jar:file:/Users/haoyi/amm!/org/apache/ivy/core/settings/ivysettings.xml
:: resolving dependencies :: com.chuusai#shapeless_2.11-caller;working
	confs: [default]
	found com.chuusai#shapeless_2.11;2.3.0-RC4 in central
	found org.scala-lang#scala-library;2.11.7 in chain-resolver
	[2.11.7] org.scala-lang#scala-library;2.11.7
	found org.typelevel#macro-compat_2.11;1.1.1 in central
	found org.scala-lang#scala-compiler;2.11.7 in chain-resolver
	found org.scala-lang#scala-reflect;2.11.7 in chain-resolver
	[2.11.7] org.scala-lang#scala-reflect;2.11.7
	found org.scala-lang.modules#scala-xml_2.11;1.0.4 in chain-resolver
	found org.scala-lang.modules#scala-parser-combinators_2.11;1.0.4 in chain-resolver

@ import shapeless._
import shapeless._
@ import syntax.singleton._
import syntax.singleton._
@ "bar".narrow
res3: bar = "bar"
@
```
